gammapy/spectrum/tests/test_flux_point.py

```diff
@@ -146,135 +146,89 @@ def test_compute_flux_points_dnde_exp(method):
     desired = model(e_ref)
     assert_quantity_allclose(actual, desired, rtol=1e-8)
 
-
-@requires_data('gammapy-extra')
-@requires_dependency('sherpa')
-@requires_dependency('matplotlib')
-@requires_dependency('scipy')
-@pytest.mark.parametrize('config', ['pl', 'ecpl'])
-def test_flux_points(config):
-    # TODO: replace this with a simple test case in a fixture
+@pytest.fixture(scope='session')
+def obs():
     filename = '$GAMMAPY_EXTRA/datasets/hess-crab4_pha/pha_obs23523.fits'
     obs = SpectrumObservation.read(filename)
+    return obs
+
+@pytest.fixture(scope='session')
+def model():
+    model = PowerLaw()
+    fit = SpectrumFit(obs(), model)
+    fit.fit()
+    fit.est_errors()
+    return fit.result[0].model
+
+@pytest.fixture(scope='session')
+def seg():
     ebounds = [0.3, 1, 30] * u.TeV
-    seg = SpectrumEnergyGroupMaker(obs=obs)
-    seg.compute_groups_fixed(ebounds=ebounds)
-
-    if config == 'pl':
-        config = dict(
-            model=PowerLaw(
-                index=Quantity(2, ''),
-                amplitude=Quantity(1e-11, 'm-2 s-1 TeV-1'),
-                reference=Quantity(1, 'TeV')
-            ),
-            obs=obs,
-            seg=seg,
-            dnde=6.523663e-11,
-            dnde_err=1.12954e-11,
-            dnde_errn=1.076992e-7,
-            dnde_errp=1.188907e-7,
-            dnde_ul=9.023994e-11,
-            res=-0.1126,
-            res_err=0.1536,
-        )
-    elif config == 'ecpl':
-        config = dict(
-            model=ExponentialCutoffPowerLaw(
```

Comment:
You are removing the ECPL test case. Why?

I think we should try to keep this.

Note that we already had bugs that our code worked for PowerLaw, but not for other spectral models such as ExposnentialCutoffPowerLaw:
https://github.com/gammapy/gammapy/issues/1219

It's OK to refactor and clean up the tests, but a spectral point for ECPL wiht an assert (if possible the existing test case) should remain.

Reply:
I agree that it is important. Are we sure about the values that ECPL is supposed to return?

Reply:
These aren't really very well verified results. So as long as you do add an ECPL test case and show that the points come out different than the PL model assumption I'm OK. Maybe leave the same / similar test case as we had and results should be similar again?